### ISSUE TITLE
Package requests==2.32.0 fails to import package ssl is not available

### ISSUE THREAD

**Issue by larsmoa** [whitphx/stlite#918]
The latest release of [requests (2.32.x)](https://pypi.org/project/requests/2.32.2/) fails to work with error `TypeError: Can't create an SSLContext object without an ssl module`.

### Minimal reproducible example: 

streamlit_app.py:
```
import requests
```

requirements.txt:
```
requests==2.32.2
```

Available on [share.stlite.net](https://share.stlite.net/#!ChBzdHJlYW1saXRfYXBwLnB5EiUKEHN0cmVhbWxpdF9hcHAucHkSEQoPaW1wb3J0IHJlcXVlc3RzGhByZXF1ZXN0cz09Mi4zMi4y)

### Discussion

I've done some investigation and this seems to happen because of the code that patches `requests` (https://github.com/whitphx/stlite/blob/d8d376bc13f01aac8546fe576cb62efa84921ef1/packages/kernel/py/stlite-server/stlite_server/bootstrap.py#L153):

- pyodide_http.patch_all() calls patch_requests()
- patch_requests() imports "from requests.adapters import BaseAdapter"
- requests.adapters imports "from urllib3.util.ssl_ import create_urllib3_context"
   and calls create_urllib3_context()
- create_urllib3_context() throws exception from this code snippet:
   ```
   if SSLContext is None:
       raise TypeError("Can't create an SSLContext object without an ssl module")
   ```
- This happens because there SSLContext is initialized as None in the "static initializer"
   on import
- SSLContext is overriden by code snippet from ssl_.py:
   ```
   try:  # Do we have ssl at all?
       import ssl
       from ssl import (  # type: ignore[assignment]
           CERT_REQUIRED,
           HAS_NEVER_CHECK_COMMON_NAME,
           OP_NO_COMPRESSION,
           OP_NO_TICKET,
           OPENSSL_VERSION,
           OPENSSL_VERSION_NUMBER,
           PROTOCOL_TLS,
           PROTOCOL_TLS_CLIENT,
           OP_NO_SSLv2,
           OP_NO_SSLv3,
           SSLContext,
           TLSVersion,
       )
       ...
   ```

Since ssl is not installed, the import fails.

**Comment by whitphx** [comment]
Thank you for reporting this.

You can workaround it by adding `ssl` to the requirements list explicitly.

Let me consider how we should fix it for a while.
Pyodide made `ssl` optional (https://pyodide.org/en/stable/usage/wasm-constraints.html) so you need to add it to the requirements list to use, then the issue is whether we should install it automatically for `requests` or leave it as-is only documenting it.

**Comment by whitphx** [comment]
@aotombielecki Thanks for the catch! It's been fixed now 👍 
Just updating the stlite version to 0.55.1 from 0.31.0 solved it somehow.

**Comment by larsmoa** [comment]
This seems to be resolved with requests==2.32.3:
> Fixed issue where Requests started failing to run on Python versions compiled without the ssl module. (#6724)
([Changelog](https://github.com/psf/requests/blob/main/HISTORY.md#2323-2024-05-29))

Suggesting to close this issue as won't fix.

**Comment by whitphx** [comment]
Thanks!
